src/proxy/routes-engine/rules/navigationFallback.ts

```diff
@@ -0,0 +1,87 @@
+import http from "http";
+import fs from "fs";
+import path from "path";
+import globalyzer from "globalyzer";
+import globrex from "globrex";
+
+// See: https://docs.microsoft.com/en-us/azure/static-web-apps/configuration#fallback-routes
+
+export const navigationFallback = async (
+  req: http.IncomingMessage,
+  _res: http.ServerResponse,
+  navigationFallback: SWAConfigFileNavigationFallback
+) => {
+  let originlUrl = req.url;
+
+  // don't process .auth or api requests
+  if (originlUrl?.startsWith("/.auth") || originlUrl?.startsWith("/api")) {
+    return;
+  }
+
+  // exit if no rewrite rule provided
+  if (!navigationFallback.rewrite) {
```

Comment:
```suggestion
  if (!navigationFallback?.rewrite) {
```

I got a null ref here

Reply:
Do we have the specs for the routes.json config? I'd love to know what property is mandatory and which is not.